LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/ChangeTrackingProperty.cs

```diff
@@ -0,0 +1,78 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace LoRaWan.NetworkServer
+{
+    using System;
+    using System.Collections.Generic;
+
+    /// <summary>
+    /// Primitive that keep tracks of changes
+    /// </summary>
+    /// <typeparam name="T">The underlying type that we keep track of. Must implement <see cref="IEqualityComparer{T}"/></typeparam>
+    public class ChangeTrackingProperty<T> : IChangeTrackingProperty
+    {
+        T current;
+        T original;
+
+        public ChangeTrackingProperty(string propertyName)
+        {
+            this.PropertyName = propertyName;
+        }
+
+        public ChangeTrackingProperty(string propertyName, T value)
+        {
+            this.PropertyName = propertyName;
+            this.current = this.original = value;
+        }
+
+        /// <summary>
+        /// Gets the property name
+        /// </summary>
+        public string PropertyName { get;  }
+
+        /// <summary>
+        /// Gets the value that must be persisted in twin collection
+        /// </summary>
+        object IChangeTrackingProperty.Value
+        {
+            get
+            {
+                if (typeof(T).IsEnum)
```

Comment:
You can cache this when creating the type, and avoid the multiple calls to typeof(T)